aws-android-sdk-s3/src/main/java/com/amazonaws/mobileconnectors/s3/transferutility/TransferUtility.java

```diff
@@ -551,20 +553,21 @@ public TransferObserver upload(String key, File file, ObjectMetadata metadata,
      * @param metadata The S3 metadata to associate with this object
      * @param cannedAcl The canned ACL to associate with this object
      * @param listener a listener to attach to transfer observer.
+     * @param connectionType The connection type to use for the transfer.
      * @return A TransferObserver used to track upload progress and state
      */
     public TransferObserver upload(String bucket, String key, File file, ObjectMetadata metadata,
```

Comment:
This would be a breaking API change for developers. Instead can you add this option through the TransferUtilityOptions object? See https://github.com/aws-amplify/aws-sdk-android/blob/master/aws-android-sdk-s3/src/main/java/com/amazonaws/mobileconnectors/s3/transferutility/TransferUtilityOptions.java. You can add a new data member with getter and setter and a default value.

Reply:
Thank you @nasdf for addressing the comments. I will be testing your changes this week and I will have an update based on how the testing goes.